File: sample/java/src/main/java/com/bestseller/api/entity/TrackingEventEntity.java
package com.bestseller.api.entity;

import org.springframework.data.annotation.Id;
import org.springframework.data.relational.core.mapping.Column;
import org.springframework.data.relational.core.mapping.Table;

import java.time.Instant;

@Table("tracking_events")
public record TrackingEventEntity(
    @Id
    @Column("id")
    Integer id,

    @Column("tracking_no")
    String trackingNo,

    @Column("timestamp")
    Instant timestamp,

    @Column("location")
    String location,

    @Column("status")
    String status,

    @Column("description")
    String description,

    @Column("created_at")
    Instant createdAt
) {
}

File: sample/dotnet/BestsellerApi.ApiService/Entities/TrackingEventEntity.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace BestsellerApi.ApiService.Entities;

[Table("tracking_events")]
public class TrackingEventEntity
{
    [Key]
    [Column("id")]
    public int Id { get; set; }

    [Required]
    [Column("tracking_no")]
    public string TrackingNo { get; set; } = string.Empty;

    [Required]
    [Column("timestamp")]
    public DateTimeOffset Timestamp { get; set; }

    [Required]
    [Column("location")]
    public string Location { get; set; } = string.Empty;

    [Required]
    [Column("status")]
    public string Status { get; set; } = string.Empty;

    [Column("description")]
    public string? Description { get; set; }

    [Column("created_at")]
    public DateTimeOffset CreatedAt { get; set; }
}
